src/engine/input/key_mapping/InputActionItem.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using Godot;
+
+/// <summary>
+///   Defines one specific input.
+///   Each InputActionItem has <see cref="InputEventItem">InputEventItems</see> associated with it.
+/// </summary>
+public class InputActionItem : VBoxContainer
+{
+    [Export]
+    public NodePath AddInputEventPath;
+
+    [Export]
+    public NodePath InputActionHeaderPath;
+
+    [Export]
+    public NodePath InputEventsContainerPath;
+
+    /// <summary>
+    ///   The group in which this action is defined.
+    /// </summary>
+    internal InputGroupItem AssociatedGroup;
+
+    private Label inputActionHeader;
+    private HBoxContainer inputEventsContainer;
+    private Button addInputEvent;
+
+    /// <summary>
+    ///   The godot specific action name
+    /// </summary>
+    /// <example>
+    ///   g_move_left, g_zoom_in
+    /// </example>
+    public string InputName { get; set; }
+
+    /// <summary>
+    ///   The string presented to the user
+    /// </summary>
+    /// <example>
+    ///   Move left, Zoom in
+    /// </example>
+    public string DisplayName { get; set; }
+
+    /// <summary>
+    ///   All the associated inputs executing this action
+    /// </summary>
+    public ObservableCollection<InputEventItem> Inputs { get; set; }
+
+    /// <summary>
+    ///   Sets up the inputs and adds them as children.
+    /// </summary>
+    public override void _Ready()
+    {
+        inputActionHeader = GetNode<Label>(InputActionHeaderPath);
+        inputEventsContainer = GetNode<HBoxContainer>(InputEventsContainerPath);
+        addInputEvent = GetNode<Button>(AddInputEventPath);
+
+        inputActionHeader.Text = DisplayName;
+
+        foreach (var input in Inputs)
+        {
+            input.AssociatedAction = this;
+            inputEventsContainer.AddChild(input);
+        }
+
+        inputEventsContainer.MoveChild(addInputEvent, Inputs.Count);
+
+        Inputs.CollectionChanged += OnInputsChanged;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (ReferenceEquals(null, obj))
+            return false;
+        if (ReferenceEquals(this, obj))
+            return true;
+        if (obj.GetType() != GetType())
+            return false;
+
+        return string.Equals(((InputActionItem)obj).InputName, InputName, StringComparison.InvariantCulture);
+    }
+
+    public override int GetHashCode()
+    {
+        return InputName != null ? InputName.GetHashCode() : 0;
+    }
+
+    /// <summary>
+    ///   The small + button has been pressed
+    /// </summary>
+    internal void OnAddEventButtonPressed()
+    {
+        var newInput = (InputEventItem)InputGroupList.InputEventItemScene.Instance();
+        newInput.AssociatedAction = this;
+        newInput.JustAdded = true;
+        Inputs.Add(newInput);
```

Comment:
Shouldn't newInput be added as a child here?

Reply:
Or does `OnInputsChanged` handle it?

Reply:
This happens here:
https://github.com/Revolutionary-Games/Thrive/blob/27b4b22ce91c02f48c11ab7b581039781479ead1/src/engine/input/key_mapping/InputActionItem.cs#L116-L125